### ISSUE TITLE
Blazor SWA giving a 404 when swa start is run

### ISSUE THREAD

**Issue by taylora3** [Azure/static-web-apps-cli#143]
When I run swa start in the root folder of the project and click on the link http://192.168.0.87:4280/ I get a 404 Not found page.

I created the Blazor swa from microsoft learn https://docs.microsoft.com/en-us/learn/modules/publish-app-service-static-web-app-api-dotnet/ and my code can be found here: https://github.com/taylora3/ShoppingList

**Steps to reproduce the behavior:**
1. Pull code from https://github.com/taylora3/ShoppingList
2. Open the terminal and run swa start
3. click on the link http://192.168.0.87:4280/
4. See error 404 page not found

**Expected behavior**
The static web app runs in the emulator as if it was being hosted on azure

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Edge
 - Version 89.0.774.57 (Official build) (64-bit)

**Additional context**
Site is up and running on https://white-glacier-097b97403.azurestaticapps.net/products for reference

Thanks for any help that can be provided 😃

**Comment by manekinekko** [comment]
Thank you @taylora3 for opening this issue.

@aaronpowell any idea how can we infer the runtime host from this dotnet project structure:

```
.
├── Api
│   ├── Api.csproj <-- dotnet api project
│   ├── ProductData.cs
│   ├── ProductsDelete.cs
│   ├── ProductsGet.cs
│   ├── ProductsPost.cs
│   ├── ProductsPut.cs
│   ├── Properties
│   ├── Startup.cs
│   ├── bin
│   ├── host.json <-- az function project
│   └── obj
├── Client
│   ├── App.razor
│   ├── Client.csproj <-- dotnet client project
│   ├── Pages
│   ├── Program.cs
│   ├── Properties
│   ├── Shared
│   ├── _Imports.razor
│   ├── bin
│   ├── obj
│   └── wwwroot
├── Data
│   ├── Data.csproj <-- data project (ignore)
│   ├── Product.cs
│   ├── bin
│   └── obj
├── README.md
├── **
└── ShoppingList.sln <-- VS project
```

**Comment by aaronpowell** [comment]
In the case where a workflow file isn't present yet in the repo, it might be a little tricky to attempt to determine where the client and api projects reside (a workflow file makes it obvious).

We can have it follow the pattern described in our docs, which results in the above demo'ed structure (`/Client` containing the Blazor app, `/Api` containing the Functions), but I foresee that not being viable for most usages as dotnet is really flexibly about folder structures and naming.

Here's a few thoughts:
* Have it look for the `Client` folder, if it doesn't find it, bail out and tell the user that there's no `package.json` and it doesn't know where the Blazor app is
* Have it look for a `.sln` file, then extract the project type GUID and look for one that matches a Blazor WASM project, thus using that
* Have it do a directory scan for all `.csproj` files and then scan each of those for references to the Blazor nupkg's, and if the UI ones are found, treat that as the Client project

None are perfect, they either require some user intervention or custom parsing, so it depends where the burden may need to reside.

**Comment by anthonychu** [comment]
IMHO to have `swa start` work magically is problematic. Started a discussion in #133 about this. We should investigate supporting a config file at the root of the repo #20 that contains settings needed to run the app (need to figure out how this would work for monorepos, could be single file with multiple contexts or multiple files). If you run `swa start` and no such file is available, I'm thinking we can interactively generate this file, using workflow and other inferred information as defaults.

Anyway, @taylora3. To unblock you right now, I think the command you should run from the repo root is:

```
swa start Client --api Api
``

**Comment by taylora3** [comment]
Thanks for your assistance. I'm still getting the 404 although it seems to be getting a bit further that it did before. I am getting a bit concerned that it an issue on my machine which I'll investigate further (node setup maybe??) 

There is a .github/workflows file. I tried using the --build flag to see if that would help but I got the same outcome. 

The application runs successfully via visual studio, if I do `dotnet run` in the client directory and also when deployed via Azure Static Web apps

Is there a reference Blazor SWA project that you know is working that I could look to use to try out the swa cli?

**Comment by anthonychu** [comment]
@taylora3 Sorry I think that's actually due to bad advice on my part.

Forgot this app is Blazor. I think your best bet is to start the Blazor dev server separately:

```
cd Client
dotnet watch run
```

Then in a second terminal:

```
swa start http://localhost:5000 --api api
```

**Comment by taylora3** [comment]
Thanks @anthonychu I got it running by running both client and api and then referencing the localhost dev servers.

I'm happy for this to be closed. I'd suggest adding a Blazor update to the readme to avoid confusion for other users.

This worked for me:
`
swa start http://localhost:5000 --api http://localhost:7071/
`
For me i'd like to run something like the following to get a Blazor project to build/run under the emulator.
`
swa start Client --blazor --api Api
`

**Comment by manekinekko** [comment]
I am glad to hear it worked for you @taylora3.

he CLI currently supports building an frontend app and an API backend using the `--build` flag. However, I am not sure the build process works great for Blazor. 

Out of curiosity, what's the exact build command you run to build this app?

**Comment by Azuredevmuc** [comment]
Hey @all,
thanks for this report and solving it. The workaround works fine for me, but it should be considered to be included in the docu, to make it easy to start with.
Thanks again for the work!

**Comment by olakusibe** [comment]
> @taylora3 Sorry I think that's actually due to bad advice on my part.
> 
> Forgot this app is Blazor. I think your best bet is to start the Blazor dev server separately:
> 
> ```
> cd Client
> dotnet watch run
> ```
> 
> Then in a second terminal:
> 
> ```
> swa start http://localhost:5000 --api api
> ```

This worked for me. 

Though still vith ```v0.6.0``` using ```swa start ./client-folder --api ./api-folder``` still gives a ```page 404```

**Comment by Azuredevmuc** [comment]
You also can do this now from the client directory:
`swa start http://localhost:5000 --run "dotnet watch run" --api ../Api`
Worked for me fine

**Comment by manekinekko** [comment]
@taylora3 We are going to close this issue. We've updated the readme with a few examples about this use case. Please open a new issue if needed.

Thank you.
